src/sinks/util/buffer/mod.rs

```diff
@@ -54,6 +58,9 @@ impl Buffer {
                 Compression::Zlib(level) => {
                     InnerBuffer::Zlib(ZlibEncoder::new(writer, level.as_flate2()))
                 }
+                Compression::Zstd(level) => {
+                    InnerBuffer::Zstd(ZstdEncoder::new(writer, level.into()).unwrap())
```

Comment:
Use of unwrap here... if it's not possible to panic here, can this be updated to `expect` with an explanation? [Definitely prefer to not have panics](https://github.com/vectordotdev/vector/blob/187f142ef5c28dec8e9b1ffbdfe0196acbe45804/docs/DEVELOPING.md?plain=1#L221).

Reply:
Agreed, this should be an `expect` in any case, and if it can possibly panic it should return an error instead.